spring-integration-file/src/main/java/org/springframework/integration/file/remote/session/DefaultSessionFactoryLocator.java

```diff
@@ -53,11 +53,23 @@ public DefaultSessionFactoryLocator(Map<Object, SessionFactory<F>> factories, Se
 	 * Add a session factory.
 	 * @param key the lookup key.
 	 * @param factory the factory.
+	 * @deprecated since 5.3 in favor of {@link #addSessionFactory}
 	 */
+	@Deprecated
 	public void addSessionFactory(String key, SessionFactory<F> factory) {
 		this.factories.put(key, factory);
```

Comment:
I have an opinion that deprecated API should call a new one for better coverage.
This way when we need to make a change in the target logic, a deprecated API won't be affected because it is meant to be for removal eventually anyway. 😄

Reply:
Yep, it make sense. Done